test/e2e/upgrade_test_test.go

```diff
@@ -0,0 +1,95 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+const (
+	upgradeNamespace = "upgrade-workload"
+)
+
+// runKibishiiTests runs kibishii tests on the provider.
+func runUpgradeTests(client testClient, toVeleroImage, providerName, veleroCLI, veleroNamespace, backupName, restoreName, backupLocation string,
+	useVolumeSnapshots bool, registryCredentialFile string) error {
+	oneHourTimeout, _ := context.WithTimeout(context.Background(), time.Minute*60)
+	serviceAccountName := "default"
+	if err := kibishiiPrepareBeforeBackup(client, providerName, oneHourTimeout, upgradeNamespace, serviceAccountName, registryCredentialFile); err != nil {
+		return errors.Wrapf(err, "Failed to install and prepare data for kibishii %s", upgradeNamespace)
+	}
+
+	defer func() {
+		if err := deleteNamespace(oneHourTimeout, client, upgradeNamespace, true); err != nil {
+			fmt.Println(errors.Wrapf(err, "failed to delete the namespace %q", upgradeNamespace))
+		}
+	}()
+
+	if err := veleroBackupNamespace(oneHourTimeout, veleroCLI, veleroNamespace, backupName, upgradeNamespace, backupLocation, useVolumeSnapshots); err != nil {
+		veleroBackupLogs(oneHourTimeout, veleroCLI, veleroNamespace, backupName)
+		return errors.Wrapf(err, "Failed to backup kibishii namespace %s", upgradeNamespace)
+	}
+
+	if providerName == "vsphere" && useVolumeSnapshots {
+		// Wait for uploads started by the Velero Plug-in for vSphere to complete
+		// TODO - remove after upload progress monitoring is implemented
+		fmt.Println("Waiting for vSphere uploads to complete")
+		if err := waitForVSphereUploadCompletion(oneHourTimeout, time.Hour, upgradeNamespace); err != nil {
+			return errors.Wrapf(err, "Error waiting for uploads to complete")
+		}
+	}
+	fmt.Printf("Simulating a disaster by removing namespace %s\n", upgradeNamespace)
+	if err := deleteNamespace(oneHourTimeout, client, upgradeNamespace, true); err != nil {
+		return errors.Wrapf(err, "failed to delete namespace %s", upgradeNamespace)
+	}
+	var cli string
+	var err error
+	main_tag := "main"
+	upgradeToVersion := veleroImage[strings.Index(toVeleroImage, ":")+1:]
+	if upgradeToVersion == main_tag {
+		cli = veleroCLI
+	} else {
+		cli, err = installVeleroCLI(upgradeToVersion)
+		if err != nil {
+			return errors.Wrapf(err, "Fail to install velero CLI.")
+		}
+	}
+	if err := veleroInstall(context.Background(), cli, toVeleroImage, veleroNamespace, cloudProvider, objectStoreProvider, useVolumeSnapshots,
+		cloudCredentialsFile, bslBucket, bslPrefix, bslConfig, vslConfig, crdsVersion, "", registryCredentialFile); err != nil {
+		return errors.Wrapf(err, "Failed to install velero from image %s", toVeleroImage)
+	}
+	time.Sleep(10 * time.Second)
```

Comment:
Is this necessary? It's better if we can find out the reason why we need to wait here

Reply:
It's fixed by modify install velero code.